Add static Next.js deployment guide (#535)
zeit/docs: pages/guides/deploying-static-nextjs-with-now.mdx

import Guide from '~/components/layout/guide'
import { TerminalInput } from '~/components/text/terminal'
import { InlineCode } from '~/components/text/code'
import Caption from '~/components/text/caption'
import Note from '~/components/text/note'
import { GenericLink } from '~/components/text/link'
import Card from '~/components/card'

export const meta = {

>   title: 'Create a Static Next.js Application and Deploy it with Now',

Reviewer: I think a more fitting title for all future Static Build guides would be to use the following format:
```suggestion
  title: 'Create a Static Next.js App That Builds and Deploys with Now',
```

  description:

>     'Learn how to create a static Next.js website and deploy it with ZEIT Now',

Reviewer: ```suggestion
    'Learn how to create a static Next.js website that builds and deploys automatically with ZEIT Now',
```

  published: '2019-02-28T13:49:00.860Z',
  authors: ['heruc'],
  url: '/guides/deploying-static-nextjs-with-now'
}

Next.js is a framework, created by [ZEIT](https://zeit.co), for creating production-ready, fast, and extended React applications with a lot of handy features built-in.

> In this guide, we will show you how to set up a static Next.js website and how to deploy it with [ZEIT Now](/docs/v2).

Reviewer: ```suggestion
In this guide, we will show you how to set up a static Next.js website and how to deploy it, with automatic builds in the cloud, with [ZEIT Now](/docs/v2).
```

You will need to create a [`now.json` configuration file](https://zeit.co/docs/v2/deployments/configuration) to instruct Now on how to handle your application in the build phase and when being served to visitors.

```json
{
  "version": 2,
  "name": "nextjs-static",
  "builds": [{ "src": "package.json", "use": "@now/static-build" }]
}
```

<Caption>
  A <InlineCode>now.json</InlineCode> configuration file that specifies the
  Static Builder.
</Caption>

The above `now.json` file achieves a few things:

- Defines a `version` property to ensure you are using the latest [Now 2.0 platform](/docs/v2/platform/overview) version.
- Defines a project `name` that your deployments will be sorted under and known by under Now.
- Defines a `builds` property as an array with one build step using `@now/static-build` to instruct Now to statically build the project and deploy the `dist` directory, using the `package.json` file as an entrypoint.

To ensure a fresh build of your Next.js project gets built by Now, add a `now-build` script to your `package.json` file:

```json
{
  ...
  "scripts": {
    ...
    "now-build": "next build && next export -o dist"
  }
}
```

> ### Serving custom files from site root

Reviewer: ```suggestion
### Serving Static Files
```

If your project needs to serve certain static files from the root of the website (like robots.txt or sitemap.xml), first copy those files to the `/static` folder, then add a routes configuration to the `now.json` file like this:

```json
{
  "version": 2,
  "name": "nextjs-static",
  "builds": [{ "src": "package.json", "use": "@now/static-build" }],
  "routes": [

>     { "src": "/robots.txt", "dest": "/static/robots.txt" },

Reviewer: I'm not sure if this is necessary or not. I will be able to test further in the morning, but I think we can just use the `@now/static` Builder to literally upload the files without these routes.

Author: Based on my testing, the `@now/static-build` will automatically upload all static files from the `static` directory and they will be available via the `/static` path by default. 
But if we want to serve specific static files from site root (for example, robots.txt is expected there), then it's necessary to specify the routes for those specific files. 

I guess I can clarify that in the guide.

    { "src": "/sitemap.xml", "dest": "/static/sitemap.xml" }
  ]
}
```

<Note>
  Don't forget to exclude the <InlineCode>node_modules</InlineCode> folder from
  being uploaded to Now to enable faster deployment. To do so, add a{' '}
  <GenericLink href="/guides/prevent-uploading-sourcepaths-with-nowignore/">
    .nowignore
  </GenericLink>{' '}
  file to the root of the project directory and add{' '}
  <InlineCode>node_modules</InlineCode> to it.
</Note>

> ### Deploying with Now

Reviewer: Could we add a caching section before this, in the setup step, that defines caching headers? It's a super important part of the deployment process that Now makes super easy. We can use the same ones as here: https://zeit.co/guides/deploying-nextjs-with-now/#immutable-caching

> You are now ready to deploy the app with Now:

Reviewer: Could we add more context on how to install Now? The user might be here for the first time, looking to host a Next.js project but doesn't know of Now. We can use the following: https://github.com/zeit/docs/blob/master/pages/guides/deploying-hugo-with-now.mdx#L126

<TerminalInput>now</TerminalInput>
<Caption>Deploying your Next.js app with Now CLI from a terminal.</Caption>

Once the app is deployed, you will receive a deployment URL similar to the following: https://nextjs-static-qstjrp75p.now.sh

> 

Reviewer: Would be great to add a bonus section here on initializing a ready-to-deploy Next.js static project with `now init` here. Like the following PR aims to do: https://github.com/zeit/docs/pull/548
